File: OnlineProgrammingChallenges/OnlineJudge/Contest Volumes/109/10920.py
import sys

class Point:
    def __init__(self, row = 0, col = 0):
        self.row, self.col = row, col
directions = [Point(1, 0), Point(0, -1), Point(-1, 0), Point(0, 1)]

# main
while (True):
    line = input()
    parts = line.split()
    size = int(parts[0])
    pos = int(parts[1])

    if (size == 0 or pos == 0):
        break

    row = size//2+1
    col = size//2+1
    dir = 0
    len = 1
    currPos = 1
    while currPos < pos:
        if len > pos - currPos:
            len = pos-currPos
        row += directions[dir].row*len
        col += directions[dir].col*len
        currPos += len
        dir = (dir + 1) % 4
        if dir % 2 == 0:
            len += 1

    print("Line = " + str(row) + ", " + "column = " + str(col) + ".")

File: OnlineProgrammingChallenges/OnlineJudge/Contest Volumes/109/10920/10920.cpp
#include <iostream>
#include <vector>

using namespace std;

struct point {
	int Row, Col;

	point() : Row(0), Col(0) {}
};


int main()
{
	int size, pos, row, col, dir, len, currPos;

	vector<point> directions(4);
	directions[0].Row = 1;
	directions[1].Col = -1;
	directions[2].Row = -1;
	directions[3].Col = 1;

	while (true) {
		cin >> size >> pos;
		if (size == 0 || pos == 0) { return 0; }

		row = size / 2 + 1;
		col = size / 2 + 1;
		dir = 0;
		len = 1;
		currPos = 1;

		while (currPos < pos) {
			if (len > pos - currPos) { len = pos - currPos; }
			row += directions[dir].Row * len;
			col += directions[dir].Col * len;
			currPos += len;
			dir = (dir + 1) % 4;
			if (dir % 2 == 0) {
				len += 1;
			}
		}

		cout << "Line = " << row << ", " << "column = " << col << "." << endl;
	}
}
